packages/react-jsx-highcharts/src/components/Annotation/Annotation.js

```diff
@@ -1,88 +1,62 @@
-import React, { Component, Children, cloneElement, isValidElement } from 'react';
+import React, { useRef, useEffect, useState, memo, Children, cloneElement, isValidElement } from 'react';
 import PropTypes from 'prop-types';
 import uuid from 'uuid/v4';
 import { attempt } from 'lodash-es';
 import Hidden from '../Hidden';
-import getModifiedProps from '../../utils/getModifiedProps';
 import { logModuleErrorMessage } from '../../utils/warnings';
+import useChart from '../UseChart';
 
-class Annotation extends Component {
+const Annotation = memo((props) => {
+  const { id = uuid, children, ...rest } = props;
 
-  static propTypes = {
-    getChart: PropTypes.func.isRequired // Provided by ChartProvider
-  };
+  const { getChart } = useChart();
```

Comment:
It might defeat the purpose of memoing everything, but is it possible for `useChart` to return [these methods](https://github.com/whawker/react-jsx-highcharts/blob/master/packages/react-jsx-highcharts/src/components/ChartProvider/index.js#L15-L25) directly? Instead of having to call `getChart()` to access them?

Reason being, a lot of our components use `chart.update({ name: { ...opts } })` and it would be nice if we could do

```js
const { update } = useChart()
update({ legend: { ...modifiedProps } })
```

I'm beginning to realise now (as you originally suggested) how hooks allow us to drop our HOC components. I can see the value of potentially dropping "context" completely and storing the current chart instance inside `useChart` via a `useRef`?

Reply:
> It might defeat the purpose of memoing everything, but is it possible for `useChart` to return [these methods](https://github.com/whawker/react-jsx-highcharts/blob/master/packages/react-jsx-highcharts/src/components/ChartProvider/index.js#L15-L25) directly? Instead of having to call `getChart()` to access them?
> 
> Reason being, a lot of our components use `chart.update({ name: { ...opts } })` and it would be nice if we could do
> 
> ```js
> const { update } = useChart()
> update({ legend: { ...modifiedProps } })
> ```

I think that is a good idea. Probably easiest to do when all provideChart calls are removed (soon).

> 
> I'm beginning to realise now (as you originally suggested) how hooks allow us to drop our HOC components. I can see the value of potentially dropping "context" completely and storing the current chart instance inside `useChart` via a `useRef`?

useChart still needs to get the chart instance from somewhere, so context is needed for that.